docs/3-implementing.md

```diff
@@ -66,3 +73,55 @@ plain field (`foo`) will be of type `keyword`, with no nested field.
 Despite the fact that IDs are often integers in various systems, this is not
 always the case. Since we want to make it possible to map as many data sources
 to ECS as possible, we default to using the `keyword` type for IDs.
+
+## Using non-ECS fields in your event stream
+
+The goal of ECS is to define a common set of fields to make working across various
+streams easier. Each stream will have their own particularities, however, and ECS
+cannot account for all possibilities. It's therefore important that implementers
+feel at ease adding fields and top level objects in their schema that are
```

Comment:
Why differentiate here between top level objects and fields? It's all fields.

Reply:
I think it's worthwhile to be explicit here. Otherwise I'm sure we'll be asked either question:

* "But can I create my own top level objects?"
* "But can I add fields to other ECS objects?"

I'm trying to make it clear that both are expected and ok from the get go.

I don't mind rewording a bit, though.